### PR TITLE
Bootstrap all packages on AppVeyor

### PR DESCRIPTION
**Pull request by DanielRuf**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] The commit message follows our guidelines: link tbd.
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

Bootstrap the packages on AppVeyor to prevent failing tests.

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[x] CI related changes
[ ] Documentation content changes
[ ] tooling / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by jantimon** [comment]
You can't use lerna before you install it - I guess you have to switch it.

**Comment by DanielRuf** [comment]
> You can't use lerna before you install it - I guess you have to switch it.

Oops, sorry. Accidentally switched both lines. Will force-push it.

**Comment by DanielRuf** [comment]
In general we could use Travis CI for macOS, Linux and Windows and would not have to manage two CI setups.

**Comment by DanielRuf** [comment]
Looks good now. Just some issues with the slow VMs of AppVeyor and some unit tests.

**Comment by jantimon** [comment]
Looks like some windows paths in snapshot cause those tests to fail - will look into it soon

**Comment by DanielRuf** [comment]
Let's do it in this branch here to see if it fails or not.

This is a classic issue (Windows uses backslash, the directory separator is different the right approach is to programmatically build the full path from the single elements as the separator value can be sometimes unreliable).

**Comment by DanielRuf** [comment]
See an example at https://repl.it/repls/YummyWhisperedCompilers

Strings are not really great for crossplatform tests ;-)

**Comment by DanielRuf** [comment]
Last change because I do not agree with this:
http://katafrakt.me/2017/09/16/80-characters-line-length-limit/

Which encourages unmaintainable / hard to maintain code and I use a screen which is rotated 90 degress to the right (more lines of code than empty space).

We should ecourage this by using some linter before running the tests.

**Comment by DanielRuf** [comment]
Last change because I do not agree with using longer lines which encourages unmaintainable / hard to maintain code and I use a screen which is rotated 90 degress to the right (more lines of code than empty space).

We should ecourage this by using some linter before running the tests.

**Comment by DanielRuf** [comment]
We should check if we can prevent this:
> Jest did not exit one second after the test run has completed.

**Comment by DanielRuf** [comment]
We should also check other occurrences of paths with forward slashes.

**Comment by DanielRuf** [comment]
https://github.com/Microsoft/TypeScript/pull/7702/files
https://github.com/Microsoft/TypeScript/issues/2965

**Comment by DanielRuf** [comment]
Well seems the prettier config (installed the deps locally now) mangles this into one line which is not great for maintaining it. Can we change the rule here a bit or ignore it? Becuase I think a multiline format is much better.

File: packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ class TsConfigWebpackPlugin {
 		console.warn(
 			"Couldn't find a tsconfig.json in the current working directory.\nYou can either set the configFile path explicitly or create a new config:\n  npx tsc --init"
 		);
-		return path.resolve(__dirname, '../config/tsconfig.base.json');
+		return path.resolve(__dirname, path.join('..', 'config', 'tsconfig.base.json'));
```
Comment (jantimon) [inline packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js]: This is already done by path.resolve as you can see in the following example:

https://runkit.com/embed/c6p0czttkcpf

File: packages/ts-config-webpack-plugin/test/TsConfigWebpackPlugin.test.js [hunk 2]
```diff
@@ -180,7 +180,7 @@ describe('TsConfigWebpackPlugin inside webpack context', () => {
 		compiler.run((err, stats) => {
 			console.warn = origWarn;
 			expect(Boolean(warnMessage)).toEqual(true);
-			const rootPath = path.resolve(__dirname, '..');
+			const rootPath = path.resolve(__dirname, '..').replace(/\\/g, '/');
```
Comment (jantimon) [inline packages/ts-config-webpack-plugin/test/TsConfigWebpackPlugin.test.js]: what about `path.posix.resolve(__dirname, '..');`?

https://nodejs.org/api/path.html#path_path_posix

File: packages/ts-config-webpack-plugin/test/TsConfigWebpackPlugin.test.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ describe('TsConfigWebpackPlugin standalone', () => {
 		const options = {};
 		const instance = new TsConfigWebpackPlugin(options);
 		const resolvedPath = instance.resolveTsConfigFilePath(__dirname);
-		expect(resolvedPath).toEqual(path.resolve(__dirname, '../tsconfig.json'));
+		expect(resolvedPath).toEqual(path.resolve(__dirname, path.join('..', 'tsconfig.json')).replace(/\\/g, '/'));
```
Comment (jantimon) [inline packages/ts-config-webpack-plugin/test/TsConfigWebpackPlugin.test.js]: what about `path.posix.resolve(__dirname, '../tsconfig.json'));`?

https://nodejs.org/api/path.html#path_path_posix

File: packages/ts-config-webpack-plugin/test/TsConfigWebpackPlugin.test.js (hunk 2 above)
Comment (DanielRuf) [reply packages/ts-config-webpack-plugin/test/TsConfigWebpackPlugin.test.js]: I've seen it but I am not aware of what it does and the proposed solution works and is used in more projects.

And other utility libs do it too.

Also see https://github.com/nodejs/node/issues/12298 and other related issues.

File: packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js (hunk 1 above)
Comment (DanielRuf) [reply packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js]: It produces different results depending on the platform and there should not be any platform specific separators imo.

File: packages/ts-config-webpack-plugin/test/TsConfigWebpackPlugin.test.js (hunk 3 above)
Comment (DanielRuf) [reply packages/ts-config-webpack-plugin/test/TsConfigWebpackPlugin.test.js]: See my other reply. Also the separator in the string can be problematic.

File: packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js (hunk 1 above)
Comment (jantimon) [reply packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js]: I don't understand what you mean - `path.resolve` is always platform specific.

Comment (DanielRuf) [reply packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js]: I know but path.posix may result in different outputs than you expect.

Comment (DanielRuf) [reply packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js]: https://github.com/sindresorhus/file-url/commit/c77fefb96adee8d5a7e1a42cb2c257dafc10e9e2

I still see no issues with .replace (like I already did in other projects and others do too).

Comment (jantimon) [reply packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js]: Oh I see - but we don't use it here.

```
 path.resolve(__dirname, '../config/tsconfig.base.json')
```

And it works perfectly on windows:

win7

Comment (DanielRuf) [reply packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js]: Please see the failing unit tests and the snapshots ;-)

Comment (DanielRuf) [reply packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js]: If this line here is not problematic I will revert it. But the others are.

File: packages/ts-config-webpack-plugin/test/TsConfigWebpackPlugin.test.js (hunk 2 above)
Comment (jantimon) [reply packages/ts-config-webpack-plugin/test/TsConfigWebpackPlugin.test.js]: You are right posix is not working as expected on windows:

win7_posix

Guess thats fine than 👍

File: packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js (hunk 1 above)
Comment (jantimon) [reply packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js]: 👍

Comment (DanielRuf) [reply packages/ts-config-webpack-plugin/src/TsConfigWebpackPlugin.js]: So far the original line does not product any failing tests on Windows. I have reverted it.

**Pull request merged**